tests/basic_relay.go

```diff
@@ -0,0 +1,298 @@
+	Expect(err).Should(BeNil())
+	receivedWarpMessage, err = avalancheWarp.ParseMessage(predicateBytes)
+	Expect(err).Should(BeNil())
+
+	// Check that the transaction has successful receipt status
+	txHash := block.Transactions()[0].Hash()
+	receipt, err = subnetBInfo.ChainWSClient.TransactionReceipt(ctx, txHash)
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	// Check that the transaction emits ReceiveCrossChainMessage
+	receiveCrossChainMessageLog := receipt.Logs[0]
+	var receiveEvent teleporterTestUtils.ReceiveCrossChainMessageEvent
+	err = teleporter.EVMTeleporterContractABI.UnpackIntoInterface(&receiveEvent, "ReceiveCrossChainMessage", receiveCrossChainMessageLog.Data)
+	Expect(err).Should(BeNil())
+	log.Info(fmt.Sprintf("Received teleporter message: %#v", receiveEvent))
+	Expect(receiveEvent.Message.MessageID.Uint64()).Should(Equal(teleporterMessageID.Uint64()))
+
+	log.Info("Finished sending warp message, closing down output channel")
+
+	// Cancel the command and stop the relayer
+	relayerCancel()
+	_ = relayerCmd.Wait()
+
+	//
+	// Try Relaying Already Delivered Message
+	//
+	logger := logging.NewLogger(
+		"awm-relayer",
+		logging.NewWrappedCore(
+			logging.Info,
+			os.Stdout,
+			logging.JSON.ConsoleEncoder(),
+		),
+	)
+	jsonDB, err := database.NewJSONFileStorage(logger, storageLocation, []ids.ID{subnetAInfo.BlockchainID, subnetBInfo.BlockchainID})
+	Expect(err).Should(BeNil())
+
+	// Modify the JSON database to force the relayer to re-process old blocks
+	jsonDB.Put(subnetAInfo.BlockchainID, []byte(database.LatestProcessedBlockKey), []byte("0"))
+	jsonDB.Put(subnetBInfo.BlockchainID, []byte(database.LatestProcessedBlockKey), []byte("0"))
+
+	// Run the relayer
+	relayerCmd, relayerCancel = testUtils.RunRelayerExecutable(ctx, relayerConfigPath)
+
+	// We should not receive a new block on subnet B, since the relayer should have seen the Teleporter message was already delivered
+	Consistently(newHeadsB, 10*time.Second, 500*time.Millisecond).ShouldNot(Receive())
+
+	// Cancel the command and stop the relayer
+	relayerCancel()
+	_ = relayerCmd.Wait()
+
+	//
+	// Validate Received Warp Message Values
+	//
+	Expect(receivedWarpMessage.SourceChainID).Should(Equal(subnetAInfo.BlockchainID))
+	addressedPayload, err := warpPayload.ParseAddressedPayload(receivedWarpMessage.Payload)
+	Expect(err).Should(BeNil())
+
+	receivedDestinationID, err := ids.ToID(addressedPayload.DestinationChainID.Bytes())
+	Expect(err).Should(BeNil())
+	Expect(receivedDestinationID).Should(Equal(subnetBInfo.BlockchainID))
+	Expect(addressedPayload.DestinationAddress).Should(Equal(teleporterContractAddress))
+	Expect(addressedPayload.Payload).Should(Equal(payload))
+
+	// Check that the teleporter message is correct
+	receivedTeleporterMessage, err := teleporter.UnpackTeleporterMessage(addressedPayload.Payload)
+	Expect(err).Should(BeNil())
+	Expect(*receivedTeleporterMessage).Should(Equal(teleporterMessage))
```

Comment:
can you add a log about successfully verifying destiantion received warp msg and teleporter msg